donkeycar/templates/complete.py

```diff
@@ -197,35 +199,37 @@ def drive(cfg, model_path=None, use_joystick=False, model_type=None,
     if use_joystick or cfg.USE_JOYSTICK_AS_DEFAULT:
         #modify max_throttle closer to 1.0 to have more power
         #modify steering_scale lower than 1.0 to have less responsive steering
-        if cfg.CONTROLLER_TYPE == "MM1":
-            from donkeycar.parts.robohat import RoboHATController            
-            ctr = RoboHATController(cfg)
-        elif "custom" == cfg.CONTROLLER_TYPE:
-            #
-            # custom controller created with `donkey createjs` command
-            #
-            from my_joystick import MyJoystickController
-            ctr = MyJoystickController(
-                throttle_dir=cfg.JOYSTICK_THROTTLE_DIR,
-                throttle_scale=cfg.JOYSTICK_MAX_THROTTLE,
-                steering_scale=cfg.JOYSTICK_STEERING_SCALE,
-                auto_record_on_throttle=cfg.AUTO_RECORD_ON_THROTTLE)
-            ctr.set_deadzone(cfg.JOYSTICK_DEADZONE)
+        if cfg.CONTROLLER_TYPE == "pigpio_pwm":    # an RC controllers read by GPIO pins. They typically don't have buttons
+            from donkeycar.parts.controller import RCReceiver
+            rc_steering = RCReceiver(cfg.STEERING_RC_GPIO, invert=True)
+            rc_throttle = RCReceiver(cfg.THROTTLE_RC_GPIO)
+            rc_wiper = RCReceiver(cfg.DATA_WIPER_RC_GPIO, jitter=0.05, no_action=0)
+            V.add(rc_steering, outputs=['user/angle', 'user/angle_on'])
+            V.add(rc_throttle, outputs=['user/throttle', 'user/throttle_on'])
+            V.add(rc_wiper, outputs=['user/wiper', 'user/wiper_on'])
+#            V.add(ctr, inputs=['cam/image_array'], outputs=['webcontroller/angle', 'webcontroller/throttle','user/mode', 'recording'],threaded=True)
         else:
-            from donkeycar.parts.controller import get_js_controller
-
-            ctr = get_js_controller(cfg)
-
-            if cfg.USE_NETWORKED_JS:
-                from donkeycar.parts.controller import JoyStickSub
-                netwkJs = JoyStickSub(cfg.NETWORK_JS_SERVER_IP)
-                V.add(netwkJs, threaded=True)
-                ctr.js = netwkJs
+            if cfg.CONTROLLER_TYPE == "custom":  #custom controller created with `donkey createjs` command
```

Comment:
There is a class of controllers that all need a different V.add that includes user/mode. That's the reason for the nested ifs. If you really don't like it, I can refactor as you suggest but it just means that I'll have to add a V.add for each of them rather than batching them as I've done here